### PR TITLE
[REEF-2016] - Fix CreateUriForPath methods of different IFileSystems to not append prefix if given path already contains it

### PR DESCRIPTION
**Pull request by dwaijam**
…prefix is given path already contains it

### CODE REVIEW

File: lang/cs/Org.Apache.REEF.IO.Tests/TestAzureDataLakeFileSystem.cs [hunk 1]
```diff
@@ -195,13 +195,21 @@ public void TestGetChildrenException()
...
             Uri createdUri = _fs.CreateUriForPath(dirStructure);
             Assert.Equal(createdUri, new Uri($"adl://{_context.AdlAccountName}{dirStructure}"));
         }
 
+        [Fact]
+        public void TestCreateUriForPathWithPrefix()
+        {
+            string dirStructure = FakeFileUri.AbsolutePath;
+            Uri createdUri = _fs.CreateUriForPath($"adl://{_context.AdlAccountName}{dirStructure}");
+            Assert.Equal(createdUri, new Uri($"adl://{_context.AdlAccountName}{dirStructure}"));
```
[the remark this answers is not in the document]
Comment (dwaijam) [reply lang/cs/Org.Apache.REEF.IO.Tests/TestAzureDataLakeFileSystem.cs]: fixed

File: lang/cs/Org.Apache.REEF.IO/FileSystem/AzureBlob/AzureBlockBlobFileSystem.cs [hunk 2]
```diff
@@ -216,5 +221,10 @@ public FileStatus GetFileStatus(Uri remoteFileUri)
 
             return new FileStatus(lastModifiedTime.Value.DateTime, blobReference.Properties.Length);
         }
+
+        private string GetUriPrefix()
+        {
+            return _client.BaseUri.AbsoluteUri.TrimEnd('/');
```
[the remark this answers is not in the document]
Comment (dwaijam) [reply lang/cs/Org.Apache.REEF.IO/FileSystem/AzureBlob/AzureBlockBlobFileSystem.cs]: removed

File: lang/cs/Org.Apache.REEF.IO/FileSystem/AzureBlob/AzureBlockBlobFileSystem.cs [hunk 3]
```diff
@@ -191,7 +191,12 @@ public IEnumerable<Uri> GetChildren(Uri directoryUri)
         /// <returns>The URI to the remote file</returns>
         public Uri CreateUriForPath(string path)
         {
-            return new Uri(_client.BaseUri.AbsoluteUri.TrimEnd('/') + '/' + path.Trim('/'));
+            if (path == null)
+            {
+                throw new ArgumentNullException(nameof(path), "Specified path is null");
+            }
+            var uriPrefix = GetUriPrefix();
+            return path.Contains(uriPrefix) ? new Uri(path) : new Uri(uriPrefix + '/' + path.Trim('/'));
```
[the remark this answers is not in the document]
Comment (tyclintw) [reply lang/cs/Org.Apache.REEF.IO/FileSystem/AzureBlob/AzureBlockBlobFileSystem.cs]: That's OK. It's worth creating a new object to get robust functionality.

File: lang/cs/Org.Apache.REEF.IO/FileSystem/AzureBlob/AzureBlockBlobFileSystem.cs [hunk 4]
```diff
@@ -191,7 +191,12 @@ public IEnumerable<Uri> GetChildren(Uri directoryUri)
         /// <returns>The URI to the remote file</returns>
         public Uri CreateUriForPath(string path)
```
Comment (tyclintw) [inline lang/cs/Org.Apache.REEF.IO/FileSystem/AzureBlob/AzureBlockBlobFileSystem.cs]: This needs to be re-evaluated. The CreateUriForPath is not doing any sanitization of the provided path. Blobs have very strict naming conventions for containers. All of which should be accounted for here.

File: lang/cs/Org.Apache.REEF.IO/FileSystem/AzureBlob/AzureBlockBlobFileSystem.cs [hunk 5]
```diff
@@ -191,7 +191,12 @@ public IEnumerable<Uri> GetChildren(Uri directoryUri)
...
         {
-            return new Uri(_client.BaseUri.AbsoluteUri.TrimEnd('/') + '/' + path.Trim('/'));
+            if (path == null)
+            {
+                throw new ArgumentNullException(nameof(path), "Specified path is null");
+            }
+            var uriPrefix = GetUriPrefix();
+            return path.StartsWith(uriPrefix) ? new Uri(path) : new Uri(uriPrefix + '/' + path.Trim('/'));
```
Comment (singlis) [inline lang/cs/Org.Apache.REEF.IO/FileSystem/AzureBlob/AzureBlockBlobFileSystem.cs]: Can this be new Uri(uriPrefix, path)? Also you can try to create a Uri from the path. If the path is relative it will throw a UriFormatException. If that is thrown, you can catch and retry with Uri(uriPrefix, path).

File: lang/cs/Org.Apache.REEF.IO/FileSystem/AzureBlob/AzureBlockBlobFileSystem.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (singlis) [reply lang/cs/Org.Apache.REEF.IO/FileSystem/AzureBlob/AzureBlockBlobFileSystem.cs]: Im concerned that we are changing the behavior of Uri because that is the expected behavior. You can probably just return _client.BaseUri as "file" would get removed when you append a new path (i.e. new Uri(_client.BaseUri, relativePath) will result in "http://www.fake.com/_relativePath_"

File: lang/cs/Org.Apache.REEF.IO/FileSystem/AzureBlob/AzureBlockBlobFileSystem.cs (hunk 5 above)
Comment (dwaijam) [reply lang/cs/Org.Apache.REEF.IO/FileSystem/AzureBlob/AzureBlockBlobFileSystem.cs]: Scott has suggsted to do something like this
try
{
   testUri = new Uri(path);
}
catch(System.UriFormatExecption)
{
   testUri = new Uri(GetBaseUri(), path);
}

File: lang/cs/Org.Apache.REEF.IO/FileSystem/AzureBlob/AzureBlockBlobFileSystem.cs (hunk 4 above)
Comment (sharathmalladi) [reply lang/cs/Org.Apache.REEF.IO/FileSystem/AzureBlob/AzureBlockBlobFileSystem.cs]: Added checks using Azure Storage apis

File: lang/cs/Org.Apache.REEF.IO/FileSystem/AzureBlob/AzureBlockBlobFileSystem.cs (hunk 5 above)
Comment (sharathmalladi) [reply lang/cs/Org.Apache.REEF.IO/FileSystem/AzureBlob/AzureBlockBlobFileSystem.cs]: Followed this suggestion and updated pull request.

File: lang/cs/Org.Apache.REEF.IO/FileSystem/AzureBlob/AzureBlockBlobFileSystem.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (sharathmalladi) [reply lang/cs/Org.Apache.REEF.IO/FileSystem/AzureBlob/AzureBlockBlobFileSystem.cs]: The GetUriPrefix() helper function was removed.

File: lang/cs/Org.Apache.REEF.IO.Tests/TestAzureBlockBlobFileSystem.cs [hunk 6]
```diff
@@ -127,7 +127,66 @@ public void TestDeleteDirectoryRecursive()
         public void TestCreateUriForPath()
         {
             var testContext = new TestContext();
-            Assert.Equal(FakeUri, testContext.GetAzureFileSystem().CreateUriForPath(FakeUri.LocalPath));
+            var tests = new[]
```
Comment (tyclintw) [inline lang/cs/Org.Apache.REEF.IO.Tests/TestAzureBlockBlobFileSystem.cs]: nit: IMO, it's better to make individual tests instead of creating these loops. That way the test runner can see which individual test is failing.

File: lang/cs/Org.Apache.REEF.IO.Tests/TestAzureBlockBlobFileSystem.cs [hunk 7]
```diff
@@ -127,7 +127,66 @@ public void TestDeleteDirectoryRecursive()
...
+                    try
+                    {
+                        resultUri = testContext.GetAzureFileSystem().CreateUriForPath(test.Path);
+                    }
+                    catch (Exception ex)
+                    {
+                        actualException = ex;
+                    }
+
+                    Assert.Null(actualException);
```
Comment (tyclintw) [inline lang/cs/Org.Apache.REEF.IO.Tests/TestAzureBlockBlobFileSystem.cs]: Shouldn't be testing the exception for null. Instead put an Assert.Fail inside the exception itself.

File: lang/cs/Org.Apache.REEF.IO.Tests/TestHadoopFileSystem.cs [hunk 8]
```diff
@@ -142,6 +143,20 @@ public void TestCreate()
...
+        {
+            var uri = _fileSystem.CreateUriForPath("/tmp/TestHadoop");
+            Assert.True(new Regex("hdfs://[a-z]+:\\d+/tmp/TestHadoop").Match(uri.AbsoluteUri).Success);
+        }
+
+        [Fact(Skip = "These tests need to be run in an environment with HDFS installed.")]
+        public void CreateUriForPathWithPrefix()
+        {
+            var uriString = "hdfs://localhost:9000/tmp/TestHadoop";
+            Assert.Equal(uriString, _fileSystem.CreateUriForPath(uriString).AbsoluteUri);
```
Comment (tyclintw) [inline lang/cs/Org.Apache.REEF.IO.Tests/TestHadoopFileSystem.cs]: Can we do this without calling "AbsoluteUri"? Seems like possiblity of unintended functionality.

```csharp
var uri = new Uri("hdfs://localhost:9000/tmp/TestHadoop");
Assert.Equal(uri, _fileSystem.CreateUriForPath(uri.ToString()));
```

File: lang/cs/Org.Apache.REEF.IO/FileSystem/AzureBlob/AzureBlockBlobFileSystem.cs [hunk 9]
```diff
@@ -191,7 +193,32 @@ public IEnumerable<Uri> GetChildren(Uri directoryUri)
...
+            try
+            {
+                resultUri = new Uri(path);
+            }
+            catch (UriFormatException)
+            {
+                resultUri = new Uri(_client.BaseUri, path);
+            }
+
+            string containerName = resultUri.Segments[1].Trim(UrlPathSeparator);
```
Comment (tyclintw) [inline lang/cs/Org.Apache.REEF.IO/FileSystem/AzureBlob/AzureBlockBlobFileSystem.cs]: Possible bounds exception on Segements?

File: lang/cs/Org.Apache.REEF.IO/FileSystem/AzureDataLake/AzureDataLakeFileSystem.cs [hunk 10]
```diff
@@ -201,7 +201,8 @@ public Uri CreateUriForPath(string path)
             {
                 throw new ArgumentNullException(nameof(path), "Specified path is null");
             }
-            return new Uri($"{GetUriPrefix()}/{path.TrimStart('/')}");
+            var uriPrefix = GetUriPrefix();
+            return path.StartsWith(uriPrefix) ? new Uri(path) : new Uri(uriPrefix + '/' + path.Trim('/'));
```
Comment (tyclintw) [inline lang/cs/Org.Apache.REEF.IO/FileSystem/AzureDataLake/AzureDataLakeFileSystem.cs]: Please append with the built-in Uri class instead of path separator operations.
```csharp
return path.StartsWith(uriPrefix) ? new Uri(path) : new Uri(uriPrefix, path);
```

File: lang/cs/Org.Apache.REEF.IO/FileSystem/Hadoop/HadoopFileSystem.cs [hunk 11]
```diff
@@ -67,11 +67,8 @@ public Uri CreateUriForPath(string path)
             {
                 throw new ArgumentException("null path passed in CreateUriForPath");
             }
-
-            Uri uri;
-            uri = new Uri(path);
+            var uri = path.StartsWith(_uriPrefix) ? new Uri(path) : new Uri(_uriPrefix + '/' + path.Trim('/'));
```
Comment (tyclintw) [inline lang/cs/Org.Apache.REEF.IO/FileSystem/Hadoop/HadoopFileSystem.cs]: Please append with the built-in Uri class instead of path separator operations.
```csharp
return path.StartsWith(uriPrefix) ? new Uri(path) : new Uri(uriPrefix, path);
```

File: lang/cs/Org.Apache.REEF.IO.Tests/TestAzureBlockBlobFileSystem.cs (hunk 6 above)
Comment (sharathmalladi) [reply lang/cs/Org.Apache.REEF.IO.Tests/TestAzureBlockBlobFileSystem.cs]: Done.

File: lang/cs/Org.Apache.REEF.IO.Tests/TestAzureBlockBlobFileSystem.cs (hunk 7 above)
Comment (sharathmalladi) [reply lang/cs/Org.Apache.REEF.IO.Tests/TestAzureBlockBlobFileSystem.cs]: This code was removed when I refactored the tests into individual tests. Hence no longer an issue.

File: lang/cs/Org.Apache.REEF.IO.Tests/TestHadoopFileSystem.cs (hunk 8 above)
Comment (sharathmalladi) [reply lang/cs/Org.Apache.REEF.IO.Tests/TestHadoopFileSystem.cs]: Done.

File: lang/cs/Org.Apache.REEF.IO/FileSystem/AzureBlob/AzureBlockBlobFileSystem.cs (hunk 9 above)
Comment (sharathmalladi) [reply lang/cs/Org.Apache.REEF.IO/FileSystem/AzureBlob/AzureBlockBlobFileSystem.cs]: Added check on Segments.Count() prior to this line.

File: lang/cs/Org.Apache.REEF.IO/FileSystem/AzureDataLake/AzureDataLakeFileSystem.cs (hunk 10 above)
Comment (sharathmalladi) [reply lang/cs/Org.Apache.REEF.IO/FileSystem/AzureDataLake/AzureDataLakeFileSystem.cs]: Fixed.

File: lang/cs/Org.Apache.REEF.IO/FileSystem/Hadoop/HadoopFileSystem.cs (hunk 11 above)
Comment (sharathmalladi) [reply lang/cs/Org.Apache.REEF.IO/FileSystem/Hadoop/HadoopFileSystem.cs]: Fixed.

File: lang/cs/Org.Apache.REEF.IO/FileSystem/AzureBlob/AzureBlockBlobFileSystem.cs [hunk 12]
```diff
@@ -186,12 +188,48 @@ public IEnumerable<Uri> GetChildren(Uri directoryUri)
...
+            if (resultUri.Segments.Count() < 2)
+            {
+                throw new ArgumentException("Input path must have a container name", nameof(path));
+            }
+
+            string containerName = resultUri.Segments[1].Trim(UrlPathSeparator);
+            NameValidator.ValidateContainerName(containerName);
+            NameValidator.ValidateBlobName(resultUri.PathAndQuery);
+
+            // If the last segment does not end with a '/', we require it to be a valid file name.
```
Comment (singlis) [inline lang/cs/Org.Apache.REEF.IO/FileSystem/AzureBlob/AzureBlockBlobFileSystem.cs]: You are trimming the uri of the / -- therefore wont this code be always executed?

Comment (tyclintw) [reply lang/cs/Org.Apache.REEF.IO/FileSystem/AzureBlob/AzureBlockBlobFileSystem.cs]: That trim happens for the containerName not the resultUri.

File: lang/cs/Org.Apache.REEF.IO/FileSystem/AzureDataLake/AzureDataLakeFileSystem.cs [hunk 13]
```diff
@@ -201,7 +206,24 @@ public Uri CreateUriForPath(string path)
...
+                resultUri = new Uri(path);
+            }
+            catch (UriFormatException)
+            {
+                resultUri = new Uri(new Uri(this.UriPrefix), path);
+            }
+
+            if (!resultUri.AbsoluteUri.StartsWith(this.UriPrefix))
+            {
+                throw new ArgumentException($"Given uri must begin with valid prefix ({this.UriPrefix})", nameof(path));
```
Comment (tyclintw) [inline lang/cs/Org.Apache.REEF.IO/FileSystem/AzureDataLake/AzureDataLakeFileSystem.cs]: nit: Should be "uri" in code but "URI" in text.

File: lang/cs/Org.Apache.REEF.IO/FileSystem/Hadoop/HadoopFileSystem.cs [hunk 14]
```diff
@@ -55,23 +55,34 @@ private HadoopFileSystem(HdfsCommandRunner commandRunner)
...
         /// </summary>
         /// <param name="path"></param>
         /// <returns></returns>
         public Uri CreateUriForPath(string path)
         {
-            Logger.Log(Level.Info, string.Format(CultureInfo.CurrentCulture, "CreateUriForPath with path: {0}, _uriPrefix: {1}.", path, _uriPrefix));
             if (path == null)
             {
-                throw new ArgumentException("null path passed in CreateUriForPath");
+                throw new ArgumentNullException("path");
```
Comment (tyclintw) [inline lang/cs/Org.Apache.REEF.IO/FileSystem/Hadoop/HadoopFileSystem.cs]: nameof(path)

File: lang/cs/Org.Apache.REEF.IO/FileSystem/Hadoop/HadoopFileSystem.cs [hunk 15]
```diff
@@ -55,23 +55,34 @@ private HadoopFileSystem(HdfsCommandRunner commandRunner)
...
+                uri = new Uri(path);
+            }
+            catch (UriFormatException)
+            {
+                uri = new Uri(new Uri(_uriPrefix), path);
+            }
+
+            if (!uri.AbsoluteUri.StartsWith(_uriPrefix))
+            {
+                throw new ArgumentException($"Given uri does not begin with valid prefix ({_uriPrefix})");
```
Comment (tyclintw) [inline lang/cs/Org.Apache.REEF.IO/FileSystem/Hadoop/HadoopFileSystem.cs]: nit: Should be "uri" in code but "URI" in text.

File: lang/cs/Org.Apache.REEF.IO/FileSystem/Hadoop/HadoopFileSystem.cs [hunk 16]
```diff
@@ -55,23 +55,34 @@ private HadoopFileSystem(HdfsCommandRunner commandRunner)
...
+            }
 
+            Logger.Log(Level.Verbose, string.Format(CultureInfo.CurrentCulture, "Uri {0} created in CreateUriForPath for path {1}.", uri, path));
```
Comment (tyclintw) [inline lang/cs/Org.Apache.REEF.IO/FileSystem/Hadoop/HadoopFileSystem.cs]: Can we make the logging consistent among the different methods?

File: lang/cs/Org.Apache.REEF.IO.Tests/TestAzureBlockBlobFileSystem.cs [hunk 17]
```diff
@@ -124,10 +129,51 @@ public void TestDeleteDirectoryRecursive()
         }
 
         [Fact]
-        public void TestCreateUriForPath()
+        public void TestCreateUriForAbsolutePathInvalid()
+        {
+            var testContext = new TestContext();
```
Comment (tyclintw) [inline lang/cs/Org.Apache.REEF.IO.Tests/TestAzureBlockBlobFileSystem.cs]: If this is used for all tests, we should initialize it in [InitializeClass].

File: lang/cs/Org.Apache.REEF.IO.Tests/TestAzureBlockBlobFileSystem.cs [hunk 18]
```diff
@@ -124,10 +129,51 @@ public void TestDeleteDirectoryRecursive()
...
+        }
+
+        [Fact]
+        public void TestCreateUriForRelativePathWithInvalidContainerName()
+        {
+            var testContext = new TestContext();
+
+            // Container name cannot contain a colon character.
+            string relativePath = "c:/folder1/folder2/file.txt";
+            Uri expectedUri = new Uri(BaseUri, relativePath);
```
Comment (tyclintw) [inline lang/cs/Org.Apache.REEF.IO.Tests/TestAzureBlockBlobFileSystem.cs]: expectedUri unused.

File: lang/cs/Org.Apache.REEF.IO.Tests/TestAzureBlockBlobFileSystem.cs [hunk 19]
```diff
@@ -124,10 +129,51 @@ public void TestDeleteDirectoryRecursive()
...
+        }
+
+        [Fact]
+        public void TestCreateUriForRelativePathWithContainerNameTooSmall()
+        {
+            var testContext = new TestContext();
+
+            // Container name must be atleast 3 characters.
+            string relativePath = "c1/folder1/folder2/file.txt";
+            Uri expectedUri = new Uri(BaseUri, relativePath);
```
Comment (tyclintw) [inline lang/cs/Org.Apache.REEF.IO.Tests/TestAzureBlockBlobFileSystem.cs]: expectedUri unused.

File: lang/cs/Org.Apache.REEF.IO.Tests/TestHadoopFileSystem.cs [hunk 20]
```diff
@@ -142,6 +143,27 @@ public void TestCreate()
             Assert.Throws<NotImplementedException>(() => _fileSystem.Create(GetTempUri()));
         }
 
+        [Fact(Skip = "These tests need to be run in an environment with HDFS installed.")]
+        public void CreateUriForPathNoPrefix()
+        {
+            Uri uri = _fileSystem.CreateUriForPath("/tmp/TestHadoop");
+            Assert.True(new Regex("hdfs://[a-z]+:\\d+/tmp/TestHadoop").Match(uri.AbsoluteUri).Success);
```
Comment (tyclintw) [inline lang/cs/Org.Apache.REEF.IO.Tests/TestHadoopFileSystem.cs]: Why use a regex? Isn't the baseUri explicit?

File: lang/cs/Org.Apache.REEF.IO.Tests/TestHadoopFileSystem.cs [hunk 21]
```diff
@@ -142,6 +143,27 @@ public void TestCreate()
...
+        [Fact(Skip = "These tests need to be run in an environment with HDFS installed.")]
+        public void TestCreateUriForPathInvalid()
+        {
+            Assert.Throws<ArgumentException>(() => _fileSystem.CreateUriForPath("http://www.invalidhadoopfs.com/container/folder1/file1.txt"));
+        }
+
+        [Fact(Skip = "These tests need to be run in an environment with HDFS installed.")]
+        public void CreateUriForPathWithPrefix()
+        {
+            string uriString = "hdfs://localhost:9000/tmp/TestHadoop";
```
Comment (tyclintw) [inline lang/cs/Org.Apache.REEF.IO.Tests/TestHadoopFileSystem.cs]: supernit: try to limit swap variables when possible.
```csharp
var uri = new Uri("hdfs://localhost:9000/tmp/TestHadoop");
Assert.Equal(uri, _fileSystem.CreateUriForPath(uri.ToString()));
```

File: lang/cs/Org.Apache.REEF.IO/FileSystem/AzureDataLake/AzureDataLakeFileSystem.cs (hunk 13 above)
Comment (sharathmalladi) [reply lang/cs/Org.Apache.REEF.IO/FileSystem/AzureDataLake/AzureDataLakeFileSystem.cs]: Fixed.

File: lang/cs/Org.Apache.REEF.IO/FileSystem/Hadoop/HadoopFileSystem.cs (hunk 14 above)
Comment (sharathmalladi) [reply lang/cs/Org.Apache.REEF.IO/FileSystem/Hadoop/HadoopFileSystem.cs]: Fixed.

File: lang/cs/Org.Apache.REEF.IO/FileSystem/Hadoop/HadoopFileSystem.cs (hunk 15 above)
Comment (sharathmalladi) [reply lang/cs/Org.Apache.REEF.IO/FileSystem/Hadoop/HadoopFileSystem.cs]: Fixed.

File: lang/cs/Org.Apache.REEF.IO.Tests/TestAzureBlockBlobFileSystem.cs (hunk 17 above)
Comment (sharathmalladi) [reply lang/cs/Org.Apache.REEF.IO.Tests/TestAzureBlockBlobFileSystem.cs]: Moved it to the constructor which is the TestInitialize for xunit tests.

File: lang/cs/Org.Apache.REEF.IO.Tests/TestAzureBlockBlobFileSystem.cs (hunk 19 above)
Comment (sharathmalladi) [reply lang/cs/Org.Apache.REEF.IO.Tests/TestAzureBlockBlobFileSystem.cs]: Removed.

File: lang/cs/Org.Apache.REEF.IO.Tests/TestHadoopFileSystem.cs (hunk 20 above)
Comment (sharathmalladi) [reply lang/cs/Org.Apache.REEF.IO.Tests/TestHadoopFileSystem.cs]: No, it's not. Look at GetUriPrefix() in HadoopFileSystem.cs.

File: lang/cs/Org.Apache.REEF.IO.Tests/TestHadoopFileSystem.cs (hunk 21 above)
Comment (sharathmalladi) [reply lang/cs/Org.Apache.REEF.IO.Tests/TestHadoopFileSystem.cs]: Removed uri variable.

File: lang/cs/Org.Apache.REEF.IO.Tests/TestHadoopFileSystem.cs (hunk 20 above)
Comment (tyclintw) [reply lang/cs/Org.Apache.REEF.IO.Tests/TestHadoopFileSystem.cs]: I stand corrected.

**Comment by sharathmalladi** [comment]
@markusweimer and @motus : This pull request has been approved by reviewers. Please review as needed and merge to master. Thanks!

**Comment by sharathmalladi** [comment]
Pinging again for @markusweimer or @motus to please merge this pull request.

The test failures that I see with the appveyor failures seem unrelated to this pull request.

File: lang/cs/Org.Apache.REEF.Driver/Evaluator/EvaluatorRequest.cs [hunk 22]
```diff
@@ -29,7 +29,7 @@ namespace Org.Apache.REEF.Driver.Evaluator
     internal class EvaluatorRequest : IEvaluatorRequest
     {
         internal EvaluatorRequest()
-            : this(0, 0, 1, string.Empty, Guid.NewGuid().ToString("N"), string.Empty, Enumerable.Empty<string>().ToList(), true, String.Empty)
```
Comment (markusweimer) [inline lang/cs/Org.Apache.REEF.Driver/Evaluator/EvaluatorRequest.cs]: Do the changes in this file actually change something material related to the goals of this PR? If not, please remove them and move them to a separate PR.

Comment (sharathmalladi) [reply lang/cs/Org.Apache.REEF.Driver/Evaluator/EvaluatorRequest.cs]: Correct. Unrelated to this PR. Will send a different pull request if needed.

**Comment by bgchun** [comment]
@dwaijam Could you resend this PR? I'm sorry for this inconvenience. We have to re-merge this PR since the previous commit's lost after resolving the apache gitbox issue.

**Comment by sharathmalladi** [comment]
OK. I resent a new pr: https://github.com/apache/reef/pull/1476. Please take a look at that.
